src/Backend/WASMWrapper.js

```diff
@@ -63,26 +62,71 @@ export default class WASMWrapper {
     async getMaterials(program) {
         const wasm = await this.wasm()
 
-        const pointer = new Int32Array(new Array(this.#materialsSize))
-        const buf = this.#_wasm._malloc(pointer.length * pointer.BYTES_PER_ELEMENT);
-        const heap = new Uint8Array(this.#_wasm.HEAP32.buffer, buf, pointer.length * pointer.BYTES_PER_ELEMENT)
+        const buf = wasm._malloc(this.#materialsSize * Int32Array.BYTES_PER_ELEMENT)
+        const heap = new Uint32Array(wasm.HEAP32.buffer, buf, this.#materialsSize)
 
-        wasm.ccall("dedx_get_material_list", null, ['number', 'number'], [heap.byteOffset, program])
+        wasm.ccall("dedx_get_material_list", null, ['number', 'number'], [program, heap.byteOffset])
 
         const result = new Int32Array(heap.buffer, heap.byteOffset, this.#materialsSize)
             .filter(x => x !== 0) // TODO: Once the new wasm is generated in the dev-precompiled delete this line
 
-        wasm._free(buf);
+        wasm._free(buf)
 
         const untyped = Array.from(result.subarray(0, result.indexOf(-1)))
 
         return this.getNames(untyped, 'dedx_get_material_name')
     }
 
-    getTrace(program, ion, material) {
-        return {
-            x: [10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
-            y: Array.from(new Array(10), _ => Math.random() * 100)
-        }
+    async getTrace(program, ion, material, method, plot_using){
+        if(method === 0)
+            return await this.getTraceByInterval(program,ion,material,plot_using)
+        else
+            return await this.getTraceByPoints(program,ion,material,plot_using)
+    }
+
+    async getTraceByPoints(program, ion, material, points) {
+        const wasm = await this.wasm()
+        const start = wasm.ccall('dedx_get_min_energy', 'number', ['number','number'],[program,ion])
+        const end = wasm.ccall('dedx_get_max_energy', 'number', ['number','number'],[program,ion])
```

Comment:
Can you write to the console (or somewhere else) min and max energy values ?
I suspect there is something wrong there